src/transformers/models/bamba/modular_bamba.py

```diff
@@ -965,6 +998,22 @@ def forward(
         if (input_ids is None) ^ (inputs_embeds is not None):
             raise ValueError("You must specify exactly one of input_ids or inputs_embeds")
 
+        padding_free_kwargs = (cu_seq_lens_q, cu_seq_lens_k, max_length_q, max_length_k, seq_idx)
+        num_padding_free_kwargs_used = sum(k is not None for k in padding_free_kwargs)
+        if num_padding_free_kwargs_used:
+            if num_padding_free_kwargs_used != len(padding_free_kwargs):
+                raise ValueError(
+                    "All of (cu_seq_lens_q, cu_seq_lens_k, max_length_q, max_length_k, seq_idx) must be specified for padding-free training."
+                )
+            if position_ids is None:
+                raise ValueError(
+                    "position_ids must also be specified when (cu_seq_lens_q, cu_seq_lens_k, max_length_q, max_length_k, seq_idx) are provided for padding-free training."
+                )
+            if attention_mask is not None:
+                raise ValueError(
+                    "attention_mask must be None when providing (cu_seq_lens_q, cu_seq_lens_k, max_length_q, max_length_k, seq_idx, position_ids) for padding-free training."
+                )
```

Comment:
in general this should not be exposed here !

Reply:
Where should sanity checks be placed? Or should I just remove them?